zerver/models/streams.py

```diff
@@ -385,3 +385,19 @@ def to_dict(self) -> Dict[str, Any]:
 
 def get_default_stream_groups(realm: Realm) -> QuerySet[DefaultStreamGroup]:
     return DefaultStreamGroup.objects.filter(realm=realm)
+
+
+class StreamTopic(models.Model):
+    MAX_NAME_LENGTH = 60
+
+    realm = models.ForeignKey(Realm, on_delete=CASCADE)
+    stream_id = models.ForeignKey(Stream, db_index=True, on_delete=CASCADE)
+    topic_name = models.CharField(max_length=MAX_NAME_LENGTH)
+    pinned = models.BooleanField(default=False)
+
+    class Meta:
+        unique_together = ("stream_id", "topic_name")
```

Comment:
Let's put this in a `zerver/models/topics.py` and name it `Topic`.

This table needs database indexes. Look at `UserTopic` for some inspiration, but mainly we need `(realm, stream_id, upper(topic_name))` as an index to easily query these for a given stream, and also a constraint that prevents duplicates with different `topic_name` case.